### PR TITLE
SBML import/export, parsing and more

### PR DESCRIPTION
**Pull request by bgoli**
I've updated the infix parsing and the general handling of SBML reading/writing of functions, events, rules

### CODE REVIEW

File: pysces/core2/PyscesCore2.py [hunk 1]
```diff
@@ -374,6 +374,7 @@ def setArg(self, var, value=None):
         self.args.append(var)
 
     def addFormula(self, formula):
+        print('PC2 377:', formula)
```
Comment (jmrohwer) [inline pysces/core2/PyscesCore2.py:377]: This looks like a debug print statement that should either be commented or put into an if-loop for debug mode.

File: pysces/core2/PyscesCore2.py [hunk 2]
```diff
@@ -1155,11 +1156,13 @@ def addSpecies(self):
             fullName = None
             if 'fullName' in self.__sDict__[s]:
                 fullName = self.__sDict__[s]['fullName']
+            # print(self.__sDict__)
             self.addOneSpecies(
                 name, val, fix=fix, comp=comp, amount=amount, fullName=fullName
             )
 
     def addOneFunction(self, name, args, formula):
+        print(name, formula)
```
Comment (jmrohwer) [inline pysces/core2/PyscesCore2.py:1165]: Debug print statement?

File: pysces/core2/PyscesCore2Interfaces.py [hunk 3]
```diff
@@ -831,12 +838,17 @@ def setSpecies(self):
                 s.setConstant(True)
             else:
                 s.setBoundaryCondition(False)
-            ##  print 'FUCK (%s) %s' % (spe.getName(), spe())
-            if spe() == None:
+            if spe() is None:
                 print(
-                    'Warning, species %s has not been initialised setting to 1.0'
-                    % spe.getName()
+                    'Warning, species {} has not been initialised setting to 1e-3'.format(spe.getName(), self.UNINIT_DEFAULT)
```
Comment (jmrohwer) [inline pysces/core2/PyscesCore2Interfaces.py:843]:
```python
'Warning, species {} has not been initialised setting to {}'.format(spe.getName(), self.UNINIT_DEFAULT)
```

File: pysces/core2/PyscesCore2Interfaces.py [hunk 4]
```diff
@@ -849,7 +861,31 @@ def setParameters(self):
...
             p.setName(par.name)
-            p.setValue(par())
+
+            # check for None value with no value and assign a default
+            #print('PAR', par.name, par())
+            # TODO: sort out this ZeroDivision and parameter initialisation error in a more sophisticated way, for example, evaluation assignment rules on export.
+            try:
+                if par() is None:
+                    print(
+                        'INFO: parameter \"{}\" not been defined setting to 1e-3'.format(par.getName(), self.UNINIT_DEFAULT)
```
Comment (jmrohwer) [inline pysces/core2/PyscesCore2Interfaces.py:871]:
```python
'INFO: parameter \"{}\" not been defined setting to {}'.format(par.getName(), self.UNINIT_DEFAULT)
```

**Review by jmrohwer — COMMENTED** [review]
@bgoli Picked up a couple of minor things in your latest merge. See inline comments. Let me know if we can fix as suggested.

File: pysces/core2/PyscesCore2.py (hunk 1 above)
Comment (bgoli) [reply pysces/core2/PyscesCore2.py:377]: thought I had all these, anyway comment out for now

File: pysces/core2/PyscesCore2.py (hunk 2 above)
Comment (bgoli) [reply pysces/core2/PyscesCore2.py:1165]: indeed, delete

**Comment by bgoli** [comment]
Thanks. Feel free to fix 👍

**Pull request merged**